File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/ReadHistoryEventRequest.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.UaStructuredType;
import org.opcfoundation.webserver.types.common.UaObjectId;

public class ReadHistoryEventRequest {
    private final UaObjectId objectId;
    private final UaStructuredType details;
    private final int offset;

    public ReadHistoryEventRequest(
            UaObjectId objectId,
            UaStructuredType details,
            int offset)
    {
        this.objectId = objectId;
        this.details = details;
        this.offset = offset;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public UaStructuredType getDetails() {
        return details;
    }

    public int getOffset() {
        return offset;
    }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/ReadHistoryEventRequest.ts
import { UaExtensionObject } from "opcua-webapi-ts";
import { UaObjectId } from "../../../types";

export class ReadHistoryEventRequest {
    private readonly _objectId: UaObjectId;
    private readonly _details: UaExtensionObject;
    private readonly _offset: number;

    constructor(
        objectId: UaObjectId,
        details: UaExtensionObject,
        offset: number,
    ) {
        this._objectId = objectId;
        this._details = details;
        this._offset = offset;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get details(): UaExtensionObject {
        return this._details;
    }

    get offset(): number {
        return this._offset;
    }
}